enhancements/update/channel-metadata.md

```diff
@@ -0,0 +1,221 @@
+
+Exposing channel descriptions to provide more context about the current channel or possible channel choices makes it easier for cluster administrators to select the channel that aligns with their intended update semantics.
+It reduces the chance that cluster administrators misinterpret the channel names (e.g. expecting `fast-*` to be unsupported, when in fact, it [is supported][channel-docs-fast]).
+
+Distributing the information via an API allows the update service, which is in charge of populating [the channel-specific update recommendations][cincinnati-channels], to declare the semantics it is using for each channel.
+This allows user-facing interfaces like `oc` and the web console to access the channel metadata and display it to the user without hard-coding assumptions about the update service.
+
+### Goals
+
+* Exposing a list of channels and related metadata provided by the upstream update recommendation service.
+
+### Non-Goals
+
+* Requiring all channels to be publically declared.
+
+## Proposal
+
+### Channel-list API
+
+This enhancement adds a new update service API, served from:
+
+```
+/api/upgrades_info/channels
+```
+
+Clients make [an HTTP GET][http-get] request of the endpoint.
+Clients and the update service may negotiate [HTTP authentication][http-authentication] to access non-public channels.
+[The `Accept` header][http-accept] should be set to `application/vnd.redhat.cincinnati.channels.v1+json`.
+
+The update service responds with [JSON][json], as required by [the `+json` structured syntax suffix][json-structured-syntax-suffix], with the following schema:
+
+* `channels` (required [object][json-object]), with the set of available channels.
+    Keys are channel names, like `stable-4.1`.
+    Values are [objects][json-object] with the following properties:
+
+    * `description` (optional [string][json-string]) with a paragraph of unstructured text describing the channel semantics.
+
+### In-cluster Storage
+
+This enhancement adds a new custom resource definition to `config.openshift.io`, based on the following Go types:
+
+```go
+// Channel represents an OpenShift update service channel.
+type Channel struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// spec is the desired state of the channel.
+	// +kubebuilder:validation:Required
+	// +required
+	Spec ChannelSpec `json:"spec"`
+
+	// status contains information about the channel.
+	// +optional
+	Status ChannelStatus `json:"status"`
+}
+
+// ChannelSpec is the desired state of the channel.  There are no
+// properties, because the channel custom resource just reflects data
+// collected from outside the cluster.
+type ChannelSpec struct {
+}
+
+// ChannelStatus represents the status of an OpenShift update service channel.
+type ChannelStatus struct {
+  // description is a paragraph of unstructured text describing the
+	// channel semantics.
+  // +optional
+  Description string `json:"description,omitempty"`
+}
```

Comment:
An example using these new API would be better for review as it will help to see how it will look like when used.